lib/AST/Util.cpp

```diff
@@ -67,9 +67,14 @@ bool IsEquivalent(clang::ASTContext &ctx, clang::Stmt *a, clang::Stmt *b) {
 }
 
 void CopyProvenance(clang::Stmt *from, clang::Stmt *to, StmtToIRMap &map) {
+  std::vector<std::pair<clang::Stmt *, llvm::Value *>> pairs;
   auto range{map.equal_range(from)};
   for (auto it{range.first}; it != range.second && it != map.end(); ++it) {
```

Comment:
`pairs.insert(pairs.end(), range.first, range.second);`?

Reply:
I'm not sure that would work, I've found that I both had to check `range.second` _and_ `map.end()`, otherwise it'd sometimes run out of bounds

Reply:
👍